src/tree/hist/evaluate_splits.h

```diff
@@ -303,22 +303,35 @@ class HistEvaluator {
           // forward enumeration: split at right bound of each bin
           loss_chg =
               static_cast<float>(evaluator.CalcSplitGain(*param_, nidx, fidx, GradStats{left_sum},
-                                                         GradStats{right_sum}) -
-                                 parent.root_gain);
-          split_pt = cut_val[i];  // not used for partition based
-          best.Update(loss_chg, fidx, split_pt, d_step == -1, false, left_sum, right_sum);
+                                                         GradStats{right_sum}) - parent.root_gain);
+          if (!is_secure_) {
+            split_pt = cut_val[i];  // not used for partition based
+            best.Update(loss_chg, fidx, split_pt, d_step == -1, false, left_sum, right_sum);
+          } else {
+            // secure mode: record the best split point, rather than the actual value
+            // since it is not accessible at this point (active party finding best-split)
+            best.Update(loss_chg, fidx, i, d_step == -1, false, left_sum, right_sum);
+          }
         } else {
```

Comment:
Do you think a [policy class](https://en.wikipedia.org/wiki/Modern_C%2B%2B_Design#Policy-based_design) might help here? Or maybe there are other efficient ways to handle these conditions? I'm loosing track of these conditions, considering that we have three enumeration functions:
- numeric
- partition
- one hot
Then we have three split modes:
- column
- row
- column + secure
So, in combination, 9 potential cases, and we haven't counted vector leaf yet. Need to find a better way to manage these conditions.

Reply:
agreed, will make some updates

Reply:
Another thing regarding this mode combinations: potentially with the upcoming processor interface we will be able to enable encrypted horizontal, shall we further add a row + secure mode, adding a 4th one for  
`enum class DataSplitMode : int { kRow = 0, kCol = 1, kColSecure = 2 };`? (or maybe there are better options?)

Reply:
For my preference, I would have put it in the `CommunicatorContext` configuration for whether the channel is encrypted.